### PR TITLE
Check register 14 (ps1 status register) for cd speed and CDDA 

### PR DESCRIPTION
**Pull request by kenshen112**
Fix to CD speed courtesy of Ref. Restored amount of old pcsx code for CDDA

### CODE REVIEW

File: pcsx2/CDVD/CDVD.cpp [hunk 1]
```diff
@@ -1186,6 +1186,8 @@ static void cdvdWrite04(u8 rt)
 { // NCOMMAND
 	CDVD_LOG("cdvdWrite04: NCMD %s (%x) (ParamP = %x)", nCmdName[rt], rt, cdvd.ParamP);
 
+	std::cout << "COMMAND: " << nCmdName[rt] << std::endl;
```
Comment (GovanifY) [inline pcsx2/CDVD/CDVD.cpp:1189]: You should use DevCon.WriteLn

File: pcsx2/CDVD/CDVD.h [hunk 2]
```diff
@@ -137,6 +137,7 @@ struct cdvdStruct
 	bool Spinning;    // indicates if the Cdvd is spinning or needs a spinup delay
 };
 
+extern cdvdStruct cdvd;
```
Comment (GovanifY) [inline pcsx2/CDVD/CDVD.h:140]: Why an extern? Can't you define the struct in the header directly?

File: pcsx2/CDVD/CdRom.cpp [hunk 3]
```diff
@@ -844,6 +841,26 @@ void cdrWrite1(u8 rt)
 			cdr.Mode = cdr.Param[0];
 			cdr.Ctrl |= 0x80;
 			cdr.Stat = NoIntr;
+			if (cdr.Mode && MODE_CDDA)
+			{
+				StopCdda();
+				cdvd.Type = CDVD_TYPE_CDDA;
+				if (cdr.Reading == 1)
```
Comment (GovanifY) [inline pcsx2/CDVD/CdRom.cpp:848]: why an == 1? isn't a standard if sufficient in this case or do we want the 1? If we do, why not an enum?

File: pcsx2/CDVD/CdRom.cpp [hunk 4]
```diff
@@ -844,6 +841,26 @@ void cdrWrite1(u8 rt)
...
+				{
+					StopReading();
+				}
+				if (cdr.SetlocPending)
+				{
+					memcpy(cdr.SetSectorSeek, cdr.SetSector, 4);
```
Comment (GovanifY) [inline pcsx2/CDVD/CdRom.cpp:854]: Why do you need a memcpy? Can't you simply make a static allocation or a bitmask if you only want 4 bytes?

**Review by GovanifY — CHANGES_REQUESTED** [review]

File: pcsx2/CDVD/CdRom.cpp [hunk 5]
```diff
@@ -226,7 +225,7 @@ void cdrInterrupt()
 			cdr.Result[0] = cdr.StatP;
 			cdr.Stat = Acknowledge;
 			break;
-
+
```
Comment (refractionpcsx2) [inline pcsx2/CDVD/CdRom.cpp:228]: Whitespace

File: pcsx2/CDVD/CdRom.cpp [hunk 6]
```diff
@@ -748,11 +745,22 @@ void cdrWrite1(u8 rt)
 
 			cdr.Ctrl |= 0x80;
 			cdr.Stat = NoIntr;
+			cdr.SetlocPending = 1;
 			AddIrqQueue(cdr.Cmd, 0x800); // the seek delay occurs on the next read / seek command (CdlReadS, CdlSeekL, etc)
 		}
 		break;
-
+		do_CdlPlay:
```
Comment (refractionpcsx2) [inline pcsx2/CDVD/CdRom.cpp:752]: What is this? does pcsxr have the goto in the middle of a switch? how does that even work? O_o

Comment (kenshen112) [reply pcsx2/CDVD/CdRom.cpp:752]: https://github.com/iCatButler/pcsxr/blob/62467b86871aee3d70c7445f3cb79f0858ec566e/libpcsxcore/cdrom.c#L653

**Pull request merged**